app/frontend/src/components/EditLocationMap.tsx

```diff
@@ -75,7 +75,7 @@ export default function EditLocationMap({
   };
 
   const onMove = (e: MapMouseEvent | MapTouchEvent) => {
-    centerCoords.current = e.lngLat;
+    centerCoords.current = e.lngLat.wrap();
```

Comment:
Is there another map component that requires similar treatment?

Reply:
That is a very good question that I wanted to pose in the chat as well. For the pages issue this was the only one affected but I wanted to have a look at the other components. As far as I see the user locations work though, so it is probably caught. We have to keep this in mind though when querying if something is contained in our country geodata.

Reply:
There is another one in the `EditUserLocationMap.tsx`, please add a wrap to that one too.

It seems postgis *does not* do the wrapping, so I'm working on a fix for that.

Reply:
I did wrap the calls in `EditUserLocationMap.tsx`. 

But I also realized that `displaceLngLat` for points close to the poles or to the border between the 180 and -180 longitude could result in an error if we step out of the ranges. Currently our data is saved as geometrical data, right? If we save it as geographical data, I could calculate the true latitude/longitude, but with geometrical data this I think it would be better to just cap the results of the function.

Reply:
Would it be possible to just `.wrap()` the coords when they are returned? That way there wouldn't be these issues (I imagine there's also an issue with the little handle if the point is on one side of the antimeridian, and the little drag handle is on the other?).

I can do a PR to update the backend/db, a double application of this function seems to be what we need: https://postgis.net/docs/ST_WrapX.html

It'd just be nice to have these constraints in the database, but it's annoying if we have to add them for every single column that has a point.

Reply:
I have to say all of those maths helpers in `EditUserLocationMap` feels like black magic to me that I don't understand... Can we add some reference as to how those formulae come about somehow?

Reply:
@aapeliv: 
The wrap only wraps longitude, not latitude overflow. And I am not really sure if we want the wrapping to happen there as we deal with purely geometrical shapes. Here we calculate a circle in which the user lives and calculate bounding points for it. If, let's say, we have something close to the North pole and one point would be wrapped from (90.2, 0) to (89.8, 180) I think it could lead to some funky shape interactions if we want to see if something is contained in a node as the geometric shape will not support that wrap. 

I saw this geography type that we could use, but it would mean changing the database and queries significantly, maybe not worth it:
https://postgis.net/workshops/postgis-intro/geography.html

Here it says that the drawbacks are that you have to cast types and execution speed:
https://blog.crunchydata.com/blog/postgis-and-the-geography-type

Generally, I think this is very much an edge case. But it would be nice if the map does not crash if the user wants to set their location to the south or north pole to pretend to be santa clause. :)

Reply:
I would not worry about the poles: that's not a big issue for us as people don't live within a few km (max radius) from the pole :).

In terms of `geom` vs `geog`: indeed geom is faster (I believe this comes from doing computations in the cartesian plane instead of on the spheroid/whatever defined by the Spatial Reference System: cartesian is just fine for an approximation when not close to the poles and when doing computations on short distances). It seems geog would wrap automatically, but I'm afraid of the performance penalty (we will end up doing quite a few point in polygon queries with community features). If needed, we can always cast to geog on the fly.

Fiji seems to be the only properly populated place that crosses the antimeridian, and even then, it's only one island with population of a few thousand people (looks funny on OSM: https://www.openstreetmap.org/relation/571747#map=8/-17.780/180.000).

Reply:
Ok, that makes sense to me. Although poor Fijians :( 

Then I would just include a latitude wrap for the function and would cap the latitude at 90° and -90° so it may show up a bit weird but does not crash for these cases.